ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/AbstractExplorerTest.java

```diff
@@ -196,9 +202,27 @@ protected void storeWindowHandle() {
     }
 
     protected void switchToNewWindow() {
-        for (String winHandle : driver.getWindowHandles()) {
-            driver.switchTo().window(winHandle);
+        Wait<WebDriver> wait = Locator.getFluentWait();
+        wait.until((new Function<WebDriver, Boolean>() {
```

Comment:
Can't a lambda be used here?

Reply:
Maybe, but we need to cleanup these logics in Locator then (where we use this everywhere so i took example on it).
I can clean this up later if needed.

Reply:
Actually we can't use a lambda as `com.google.common.base.Function` is not a SAM-type (single abstract method) required for lambda use despite being annotated `@FunctionalInterface`